src/mod/endpoints/mod_sofia/sofia_reg.c

```diff
@@ -205,6 +205,16 @@ void sofia_reg_truly_del_gateway(sofia_profile_t *profile)
 				}
 			}
 
+			if (gateway_ptr->sofia_private) {
+				sofia_private_free(gateway_ptr->sofia_private);
+			}
+
+			if (gateway_ptr->nh) {
```

Comment:
Move this on the statement below, it should be deleted during the state of REG_STATE_DOWN